core/src/main/java/io/questdb/cairo/CairoEngine.java

```diff
@@ -653,6 +728,35 @@ private void tryRepairTable(
         }
     }
 
+    private void unlockReaders(CharSequence tableName, Path path) {
+        // this method must be invoked only when thread has lock on writer pool
+        path.of(configuration.getRoot()).concat(tableName);
+        int plen = path.length();
+
+        final FilesFacade ff = configuration.getFilesFacade();
+        try (TxnScoreboard txnScoreboard = new TxnScoreboard(configuration)) {
+            if (txnScoreboard.ofRW(path, true) == null) {
+                // we should not be unlocking readers unless we could lock them in the first place
+                // these scenarios are possible here:
+                // 1. unlockReaders() was called spuriously, without attempting to lock them first
+                // 2. somehow readers were locked and scoreboard file existed, but it disappeared before unlock was called
+                //    2.1 the drop table operation locked readers, removed the table and then attempted to unlock something that no longer exists
+                LOG.critical().$("internal error, could not open [file=").$(path).I$();
+                return;
+            }
+            final long txn = readMaxTxn(ff, path.trimTo(plen));
+            if (txn == -2) {
+                // could not read the txn, message would be already logged
+                return;
+            }
+
+            if (txnScoreboard.getActiveReaderCount(txn) == -1 && txnScoreboard.acquireTxn(txn, false)) {
```

Comment:
What if you failed to lock because a Reader existed in another engine, when you try to unlock here, reader still exists, so `getActiveReaderCount()` still returns 0 and then the scoreboard is never unlocked.